StockTrader/src/main/resources/schema.sql

```diff
@@ -37,45 +37,48 @@ CREATE TABLE account (
 );
 
 CREATE TABLE account_transaction_log (
-    action varchar,
+    transaction_num SERIAL PRIMARY KEY,
+    action varchar(255),
     funds integer,
-    timestamp timestamp,
-    username varchar
+    timestamp bigint,
+    username varchar(255)
 );
 
-CREATE TABLE user_command_log (
-    transaction_num integer,
-    timestamp timestamp without time zone,
-    server varchar(255),
-    command varchar(255),
-    username varchar(255),
-    stock_symbol varchar(255),
-    filename varchar(255),
-    funds integer
+CREATE TABLE log_xml (
+    id SERIAL PRIMARY KEY,
+    xml_log_entry varchar,
+    user_id varchar(255)
 );
 
 
 
 CREATE OR REPLACE FUNCTION log_account_transaction() RETURNS trigger AS '
 DECLARE
     action varchar;
+    old_amount integer;
     funds integer;
 BEGIN
     -- For inserts (ie. new accounts), treat previous amount as 0.
-     IF (OLD IS NULL) THEN
-        OLD.amount = 0;
+     IF (TG_OP = ''INSERT'') THEN
+        old_amount = 0;
+     ELSE
+        old_amount = OLD.amount;
      END IF;
-     IF (OLD.amount < NEW.amount) THEN
+     IF (old_amount < NEW.amount) THEN
         action = ''add'';
-        funds = New.amount - OLD.amount;
-      ELSIF (OLD.amount > NEW.amount) THEN
+        funds = New.amount - old_amount;
+      ELSIF (old_amount > NEW.amount) THEN
         action = ''remove'';
-        funds = OLD.amount - NEW.amount;
+        funds = old_amount - NEW.amount;
       ELSE
         RETURN NULL;
       END IF;
-      INSERT INTO account_transaction_log (action, funds, timestamp, username)
```

Comment:
We should still insert into account transaction log, since we need transaction history for the display_summary command

Reply:
Wouldn't the XML dumplog_per_user file be enough? We could display the summary to the user (I assume  in a readable format on the client interface) based on the XML file data itself.

Reply:
we don't want all the other log types mixed in though, the dumplog command downloads a file, and the dumplog command is probably going to be quite slow compared to what we would want for display_summary. Think of display_summary like going to your profile page. do you want a website to download an xml file every time you go to your profile?

Reply:
Okay. What do you think we would want for display summary? It doesn't say very much on the project website.